### PR TITLE
ACS-8106: (a11y) Replace Material Tooltip with standard tooltip

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

Switch to standard browser tooltips instead of the material one:

- supported by all browsers in terms of a11y
- respects browser theming (dark,light)
- respects RTL languages (previously ADF was restricting positions of tooltips)

cleanup the irrelevant unit tests related to testing the mat tooltips

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://hyland.atlassian.net/browse/ACS-8106

### CODE REVIEW

**Review by dominikiwanekhyland — APPROVED** [review]

**Review by DaryaBalvanovich — APPROVED** [review]
Great improvement, now we have only one tooltip in ADF! :tada:
Are we planning to refactor ACA and monorepo as well?

File: lib/content-services/src/lib/search/components/search-filter/search-filter.component.html [hunk 1]
```diff
@@ -3,8 +3,7 @@
     <button *ngIf="displayResetButton && facetFiltersService.responseFacets"
             mat-button
             color="primary"
-            matTooltip="{{ 'SEARCH.FILTER.BUTTONS.RESET-ALL.TOOLTIP' | translate }}"
-            matTooltipPosition="right"
```
Comment (MichalKinas) [inline lib/content-services/src/lib/search/components/search-filter/search-filter.component.html:7]: Same here

File: lib/content-services/src/lib/search/components/search-radio/search-radio.component.html [hunk 2]
```diff
@@ -6,10 +6,7 @@
         *ngFor="let option of options"
         [value]="option.value"
         class="adf-facet-filter">
-        <div
-            matTooltip="{{ option.name | translate }}"
-            matTooltipPosition="right"
```
Comment (MichalKinas) [inline lib/content-services/src/lib/search/components/search-radio/search-radio.component.html:11]: Same here

File: lib/core/src/lib/context-menu/context-menu.spec.ts [hunk 3]
```diff
@@ -197,27 +196,5 @@ describe('ContextMenuDirective', () => {
                 ).length
             ).toBe(0);
         });
-
-        it('should show tooltip if is set', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/context-menu/context-menu.spec.ts:201]: Can we instead verify if element contains the right tooltip attribute?

File: lib/core/src/lib/form/components/widgets/amount/amount.widget.spec.ts [hunk 4]
```diff
@@ -85,23 +84,6 @@ describe('AmountWidgetComponent', () => {
             });
             fixture.detectChanges();
         });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/amount/amount.widget.spec.ts:89]: Same here for testing

File: lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.html [hunk 5]
```diff
@@ -6,10 +6,8 @@
                   [disabled]="field.readOnly || readOnly"
                   [(ngModel)]="field.value"
                   (ngModelChange)="onFieldChanged(field)"
-                  [matTooltip]="field.tooltip"
-                  (click)="markAsTouched()"
-                  matTooltipPosition="right"
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.html:11]: Same here for position

File: lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.spec.ts [hunk 6]
```diff
@@ -96,32 +94,5 @@ describe('CheckboxWidgetComponent', () => {
...
-        describe('when tooltip is set', () => {
-            beforeEach(() => {
-                widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                    type: FormFieldTypes.BOOLEAN,
-                    tooltip: 'my custom tooltip'
-                });
-                fixture.detectChanges();
-            });
-
-            it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.spec.ts:109]: Same here for testing

File: lib/core/src/lib/form/components/widgets/date-time/date-time.widget.spec.ts [hunk 7]
```diff
@@ -233,33 +231,6 @@ describe('DateTimeWidgetComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.DATETIME,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/date-time/date-time.widget.spec.ts:245]: Same here for testing

File: lib/core/src/lib/form/components/widgets/decimal/decimal.component.spec.ts [hunk 8]
```diff
@@ -45,33 +44,6 @@ describe('DecimalComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.DECIMAL,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/decimal/decimal.component.spec.ts:57]: Same here for testing

File: lib/core/src/lib/form/components/widgets/multiline-text/multiline-text.widget.spec.ts [hunk 9]
```diff
@@ -40,33 +39,6 @@ describe('MultilineTextWidgetComponentComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.MULTILINE_TEXT,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/multiline-text/multiline-text.widget.spec.ts:52]: Same here for testing

File: lib/core/src/lib/form/components/widgets/number/number.widget.spec.ts [hunk 10]
```diff
@@ -42,33 +41,6 @@ describe('NumberWidgetComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.NUMBER,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/number/number.widget.spec.ts:54]: Same here for testing

File: lib/core/src/lib/form/components/widgets/text/text.widget.spec.ts [hunk 11]
```diff
@@ -153,33 +152,6 @@ describe('TextWidgetComponent', () => {
...
-        describe('when tooltip is set', () => {
-            beforeEach(() => {
-                widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                    type: FormFieldTypes.TEXT,
-                    tooltip: 'my custom tooltip'
-                });
-                fixture.detectChanges();
-            });
-
-            it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/core/src/lib/form/components/widgets/text/text.widget.spec.ts:165]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts [hunk 12]
```diff
@@ -926,37 +921,4 @@ describe('AttachFileCloudWidgetComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.UPLOAD,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts:939]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.spec.ts [hunk 13]
```diff
@@ -433,34 +427,6 @@ describe('DateWidgetComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.DATE,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.spec.ts:445]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.spec.ts [hunk 14]
```diff
@@ -259,38 +258,6 @@ describe('DropdownCloudWidgetComponent', () => {
...
-    describe('when tooltip is set', () => {
-        beforeEach(() => {
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.DROPDOWN,
-                tooltip: 'my custom tooltip'
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.spec.ts:271]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/group/group-cloud.widget.spec.ts [hunk 15]
```diff
@@ -67,31 +66,6 @@ describe('GroupCloudWidgetComponent', () => {
             });
             fixture.detectChanges();
         });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/group/group-cloud.widget.spec.ts:71]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/people/people-cloud.widget.spec.ts [hunk 16]
```diff
@@ -95,31 +94,6 @@ describe('PeopleCloudWidgetComponent', () => {
             });
             fixture.detectChanges();
         });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/people/people-cloud.widget.spec.ts:99]: Same here for testing

File: lib/process-services-cloud/src/lib/form/components/widgets/radio-buttons/radio-buttons-cloud.widget.spec.ts [hunk 17]
```diff
@@ -204,34 +203,4 @@ describe('RadioButtonsCloudWidgetComponent', () => {
...
-            widget.field = new FormFieldModel(new FormModel({ taskId: '<id>' }), {
-                type: FormFieldTypes.RADIO_BUTTONS,
-                tooltip: 'my custom tooltip',
-                optionType: 'manual',
-                options: restOption
-            });
-            fixture.detectChanges();
-        });
-
-        it('should show tooltip', async () => {
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/radio-buttons/radio-buttons-cloud.widget.spec.ts:219]: Same here for testing

File: lib/process-services/src/lib/form/widgets/radio-buttons/radio-buttons.widget.spec.ts [hunk 18]
```diff
@@ -227,26 +226,6 @@ describe('RadioButtonsWidgetComponent', () => {
             expect(radioButtonWidget.field.isValid).toBe(true);
         });
 
-        it('should display tooltip when tooltip is set', async () => {
```
Comment (MichalKinas) [inline lib/process-services/src/lib/form/widgets/radio-buttons/radio-buttons.widget.spec.ts:230]: Same here for testing

**Comment by DenysVuika** [comment]
@MichalKinas I've did a quick smoke with linked ACA to check the tooltips are working, i.e.

Screenshot 2024-07-02 at 9 18 48 AM

Screenshot 2024-07-02 at 9 21 17 AM

For the unit tests, that's a good point, I will restore the unit tests and check the attribute instead

**Comment by MichalKinas** [comment]
> @MichalKinas I've did a quick smoke with linked ACA to check the tooltips are working, i.e.
> 
> Screenshot 2024-07-02 at 9 18 48 AM Screenshot 2024-07-02 at 9 21 17 AM
> For the unit tests, that's a good point, I will restore the unit tests and check the attribute instead

Sounds good thanks!

File: lib/content-services/src/lib/search/components/search-chip-list/search-chip-list.component.html [hunk 19]
```diff
@@ -5,8 +5,7 @@
             color="primary"
             selected
             class="adf-search-chip-list-item"
-            matTooltip="{{ 'SEARCH.FILTER.BUTTONS.CLEAR-ALL.TOOLTIP' | translate }}"
-            matTooltipPosition="right"
```
[the remark this answers is not in the document]
Comment (DenysVuika) [reply lib/content-services/src/lib/search/components/search-chip-list/search-chip-list.component.html:9]: should be working fine, the `title` is the DOM attribute that all browsers always respect

File: lib/content-services/src/lib/search/components/search-filter/search-filter.component.html (hunk 1 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/search/components/search-filter/search-filter.component.html:7]: yes, added some screenshots

File: lib/content-services/src/lib/search/components/search-radio/search-radio.component.html (hunk 2 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/search/components/search-radio/search-radio.component.html:11]: it's the same behavior as before but without extra lib

File: lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.html (hunk 5 above)
Comment (DenysVuika) [reply lib/core/src/lib/form/components/widgets/checkbox/checkbox.widget.html:11]: the position should not be there, it's the browser who decides on positioning (i.e. scrolling, RTL layout, etc)

**Review by MichalKinas — APPROVED** [review]

File: lib/content-services/src/lib/search/components/search-filter/search-filter.component.html (hunk 1 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/search/components/search-filter/search-filter.component.html:7]: should be working fine, added a few screenshots from the ACA smoke tests

**Pull request merged**